api/search.go

```diff
@@ -279,89 +282,71 @@ func (api *SearchAPI) deleteSearchIndex(w http.ResponseWriter, r *http.Request)
 	logData := log.Data{"instance_id": instanceID, "dimension": dimension}
 	auditParams := common.Params{"instance_id": instanceID, "dimension": dimension}
 
-	ctx := r.Context()
+	log.InfoCtx(ctx, "deleteSearchIndex endpoint: attempting to delete search index", logData)
 
-	if err := api.auditor.Record(ctx, models.AuditTaskDeleteIndex, models.AuditActionAttempted, auditParams); err != nil {
-		handleAuditingFailure(w, ctx, models.AuditTaskDeleteIndex, models.AuditActionAttempted, err, logData)
-		return
-	}
-
-	status, err := api.elasticsearch.DeleteSearchIndex(r.Context(), instanceID, dimension)
+	status, err := api.elasticsearch.DeleteSearchIndex(ctx, instanceID, dimension)
 	logData["status"] = status
 	if err != nil {
-		if err := api.auditor.Record(r.Context(), models.AuditTaskDeleteIndex, models.AuditActionUnsuccessful, auditParams); err != nil {
-			handleAuditingFailure(w, ctx, models.AuditTaskDeleteIndex, models.AuditActionUnsuccessful, err, logData)
+		if auditError := api.auditor.Record(ctx, models.AuditTaskDeleteIndex, models.AuditActionUnsuccessful, auditParams); auditError != nil {
+			http.Error(w, internalError, http.StatusInternalServerError)
 			return
 		}
-		setErrorCode(w, err, "failed to delete index")
-		return
-	}
 
-	if err := api.auditor.Record(r.Context(), models.AuditTaskDeleteIndex, models.AuditActionSuccessful, auditParams); err != nil {
-		handleAuditingFailure(w, ctx, models.AuditTaskDeleteIndex, models.AuditActionSuccessful, err, logData)
+		setErrorCode(ctx, w, err)
 		return
 	}
 
+	api.auditor.Record(ctx, models.AuditTaskDeleteIndex, models.AuditActionSuccessful, auditParams)
+
 	setJSONContentType(w)
 	w.WriteHeader(http.StatusOK)
 
-	log.Info("index deleted", logData)
+	log.InfoCtx(ctx, "deleteSearchIndex endpoint: search index deleted", logData)
 }
 
 func setJSONContentType(w http.ResponseWriter) {
 	w.Header().Set("Content-Type", "application/json")
 }
 
-func setErrorCode(w http.ResponseWriter, err error, errorContext string) {
-
-	if strings.HasPrefix(err.Error(), "invalid response: 401") {
-		http.Error(w, err.Error(), http.StatusUnauthorized)
-	}
-
-	if matches := reNotFound.FindStringSubmatch(err.Error()); len(matches) > 0 {
-		err = errors.New(matches[1])
-	}
-
-	log.ErrorC(errorContext, err, nil)
+const (
```

Comment:
feels like the consts should be at the top of the file with the others (or the error handling should be in a separate file). they're a bit buried here